File: app/src/main/java/com/falcon/restaurants/screens/meal/FilterMealsUseCase.java
package com.falcon.restaurants.screens.meal;

import com.falcon.restaurants.room.meal.Meal;
import com.falcon.restaurants.base.FilterInterface;
import javax.inject.Inject;

public class FilterMealsUseCase extends FilterInterface<Meal> {

    @Inject
    public FilterMealsUseCase() {}

    @Override
    public boolean isQueryFound(String query, Meal current) {

        return  current.getName().toLowerCase().contains(query.toLowerCase()) /* ||
                current.getDetails().toLowerCase().contains(query.toLowerCase())*/ ;
    }

}

File: app/src/main/java/com/falcon/restaurants/screens/meal/FilterMealsUseCase.kt
package com.falcon.restaurants.screens.meal
import com.falcon.restaurants.room.meal.Meal
import com.falcon.restaurants.base.FilterBase
import javax.inject.Inject

class FilterMealsUseCase @Inject constructor(): FilterBase<Meal>() {

    override fun isQueryFound(query: String , current: Meal): Boolean{

        return current.name.lowercase().contains(query.lowercase()) /* ||
                current.details.lowercase().contains(query.lowercase())*/
    }

}
